BLEN-124: Add plugin configuration options in settings panel

VascoPi opened the pull request:
Add plugin configuration options in the settings panel.
Changed config to addon preferences.

VascoPi committed:
BLEN-124: Add plugin configuration options in settings panel.

bnagirniak commented on extern/hdusd/addon/__init__.py:
```diff
@@ -11,9 +11,11 @@
+    "doc_url": "https://radeon-pro.github.io/RadeonProRenderDocs/en/usd_hydra/about.html",
+    "community": "https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/discussions",
+    "downloads": "https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/releases",
+    "main_web": "https://www.amd.com/en/technologies/radeon-prorender",
```
Leave URLs empty

bnagirniak commented on extern/hdusd/addon/__init__.py:
```diff
@@ -39,6 +41,70 @@
+log = logging.Log('init')
+
+
+class HDUSD_ADDON_PT_preferences(AddonPreferences):
```
Move to properties/preferences.py

bnagirniak commented on extern/hdusd/addon/ui/material.py:
```diff
@@ -714,8 +713,7 @@ class HDUSD_MATERIAL_PT_dev(HdUSD_ChildPanel):
     def poll(cls, context):
-        return
-        #return config.show_dev_settings
+        return context.preferences.addons['hdusd'].preferences.dev_tools
```
make function for `context.preferences.addons['hdusd'].preferences`

bnagirniak commented on extern/hdusd/python.cpp:
```diff
@@ -239,6 +240,14 @@ static PyObject *stage_free_func(PyObject * /*self*/, PyObject *args)
+  DLOG(INFO) << "get_temp_dir_func";
+  filesystem::path path = hdusd::get_temp_dir();
+  std::string str{path.u8string()};
+  return PyUnicode_FromString(str.c_str());
```
try directly path.u8string().c_str()

bnagirniak commented on extern/hdusd/addon/__init__.py:
```diff
@@ -39,6 +41,70 @@
+        col.separator()
+        row = col.row()
+        row.operator("wm.url_open", text="Main Site", icon='URL').url = bl_info["main_web"]
+        row.operator("wm.url_open", text="Community", icon='COMMUNITY').url = bl_info["community"]
```
comment this

VascoPi committed:
Moved HDUSD_ADDON_PT_preferences to properties/preferences.py
Cleanup code.
Added function get_addon_pref.
Added preferences.py to CMakeLists.txt.

VascoPi committed:
Improved get_temp_dir_func function.
